blockmanager.go

```diff
@@ -1151,26 +1136,14 @@ func (b *blockManager) handleNotifyMsg(notification *blockchain.Notification) {
 		// transaction are NOT removed recursively because they are still
 		// valid.
 		for _, tx := range block.Transactions()[1:] {
-			b.server.txMemPool.RemoveTransaction(tx, false)
-			b.server.txMemPool.RemoveDoubleSpends(tx)
-			b.server.txMemPool.RemoveOrphan(tx)
-			acceptedTxs := b.server.txMemPool.ProcessOrphans(tx)
+			b.txMemPool.RemoveTransaction(tx, false)
+			b.txMemPool.RemoveDoubleSpends(tx)
+			b.txMemPool.RemoveOrphan(tx)
+			b.server.TransactionConfirmed(tx)
+			acceptedTxs := b.txMemPool.ProcessOrphans(tx)
 			b.server.AnnounceNewTransactions(acceptedTxs)
```

Comment:
Along the same lines, this should also be a callback in the newly created config struct.